File: src/my_project_typescript/easy_problems_typescript/from1to50/luckyNumbersMatrix.ts
function luckyNumbers (matrix: number[][]): number[] {
    let solution: Array<number> = [];

    let getColumnMax = (matrix: number[][], j: number) => {
        let mySlide: Array<number> = [];
        for (let i = 0; i < matrix.length; i++){
            mySlide.push(matrix[i][j])
        }
        return Math.max.apply(null, mySlide)
    }

    let getRowMin = (matrix: number[][], i: number) => {
        let mySlideother: Array<number> = [];
        for (let j = 0; j < matrix[0].length; j++){
            mySlideother.push(matrix[i][j])
        }
        return Math.min.apply(null, mySlideother)
    }

    for (let i = 0; i < matrix.length; i++){
        for (let j = 0; j < matrix[0].length; j++){
            console.log( getRowMin(matrix, i), getColumnMax(matrix,j))
            if (matrix[i][j] === getRowMin(matrix, i) && matrix[i][j] === getColumnMax(matrix,j)){
                solution.push(matrix[i][j])
            }
        }
    }




    return solution

};

console.log(luckyNumbers([[1,10,4,2],[9,3,8,7],[15,16,17,12]]))

File: src/my_project/easy_problems/from101to150/lucky_numbers_matrix.py
from typing import List 

class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        solution: List[int] = list()

        def getColumn(matrix, j):
            my_slide: List[int] = list()
            for i in range(len(matrix)):
                my_slide.append(matrix[i][j])
            return my_slide 

            return None 
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                print( min([x for x in matrix[i][:]]) ,   max(getColumn(matrix, j)))
                # print(matrix[i][j])
                if matrix[i][j] == min([x for x in matrix[i][:]]) and matrix[i][j] == max(getColumn(matrix, j)): 
                    solution.append(matrix[i][j])
        return solution



solution = Solution()

print(solution.luckyNumbers(matrix = [[3,7,8],[9,11,13],[15,16,17]]))




